File: 프로그래머스/0/181843. 부분 문자열인지 확인하기/부분 문자열인지 확인하기.java
class Solution {
    public int solution(String my_string, String target) {
        int len1 = my_string.length();
        int len2 = target.length();
        for(int i = 0; i<= len1 - len2; i++){
            boolean match = true;
            for(int j = 0; j< len2; j++){
                if (my_string.charAt(i + j) != target.charAt(j)) {
                    match = false;
                    break;
                }
            }
             if (match) return 1;
        }
        return 0;
    }
}

File: 프로그래머스/0/181843. 부분 문자열인지 확인하기/부분 문자열인지 확인하기.py
def solution(my_string, target):
    start = 0
    end = len(target)

    if end > len(my_string):
        return 0
    
    while end <= len(my_string):
        if my_string[start:end] == target:
            return 1
        start += 1
        end += 1
        
    return 0